fontforge/splinestroke.c

```diff
@@ -1518,16 +1778,304 @@ static void MiterJoin(JoinParams *jpp) {
+
+    p_fm = jpp->cur->last->me;
+    p_to = jpp->oxy;
+    ut_to = jpp->no_to->utanvec;
+
+    /* Arcs is the only join style with parameters beyond the position and
+     * slope at the offset spline endpoints. The "from" spline is available
+     * as jpp->cur->last->from but the "to" spline is not yet built.
+     *
+     * We could calculate its curvature by the curvature of the
+     * source spline and the nib point, but the point fitting/estimation
+     * process may yield a visibly different result. We could instead
+     * reorganize the code so that joins are retroactively applied, but
+     * that would both take effort and leave the code less readable.
+     * Instead we build the "to" spline here (when necessary), extract the
+     * parameters, and delete it.
+     *
+     * Regardless if either "join point" is on a cusp the curvature of the
+     * source spline has been "overwhelmed" to the opposite sign by the 
+     * curvature of the nib. In that case we just use the unadjusted source
+     * curvature, which (because the radius is relatively small) will either
+     * be adjusted to an appropriate larger value or subject to the Round
+     * join fallback. 
+     */
+
+    t = SplineStrokeNextT(jpp->c, jpp->s, 0.0, jpp->ccw_to, &ut_to, &curved,
+                          jpp->is_right, -1);
+    ut_to = jpp->no_to->utanvec; // Set back after call
+
+    if ( curved ) {
+	if ( OffsetOnCuspAt(jpp->c, jpp->s, 0.0, jpp->no_to,
+	                    jpp->is_right, jpp->ccw_to) ) {
+	    // printf("Outgoing on cusp - using source curvature\n");
+	    K_to = SplineEndCurvature(jpp->s, false);
+	} else {
+	    tmp_st = SplinePointCreate(p_to.x, p_to.y);
+	    tmp_end = TraceAndFitSpline(jpp->c, jpp->s, 0.0, t, tmp_st, -1,
+	                                jpp->is_right, false);
+	    K_to = SplineEndCurvature(tmp_st->next, false);
+	    while ( tmp_st!=tmp_end ) {
+		tmp_st = tmp_st->next->to;
+		free(tmp_st->prev->from);
+		free(tmp_st->prev);
+	    }
+	    free(tmp_end);
+	}
+    } else {
+	// If the spline will be drawn by a nib point as opposed to a nib curve
+	// then the curvature will just be that of the source spline.
+	K_to = SplineEndCurvature(jpp->s, false);
+    }
+    if ( OffsetOnCuspAt(jpp->c, jpp->s->from->prev, 1.0, NULL,
+	                jpp->is_right, jpp->ccw_fm) ) {
+	//printf("Incoming on cusp - using source curvature\n");
+	K_fm = SplineEndCurvature(jpp->s->from->prev, false);
+    } else
+	K_fm = SplineEndCurvature(jpp->cur->last->prev, true);
+
+    fsw_fm = FalseStrokeWidth(jpp->c, BPNeg(jpp->ut_fm));
+    fsw_to = FalseStrokeWidth(jpp->c, BPNeg(jpp->no_to->utanvec));
+    fsw_avg = (fsw_fm + fsw_to) / 2.0;
+
+    // printf(" K_fm: %.15lf, K_to: %.15lf\n", K_fm, K_to);
+    if ( K_fm==0 && K_to==0 ) {
+	MiterJoin(jpp);
+	return;
+    }
+
+    // Calculate derived parameters
+    neg_fm = (K_fm < 0);
+    neg_to = (K_to < 0);
+
+    if ( K_fm!=0.0 ) {
```

Comment:
Logic seems ok, but it's still raising this (probably false) warning:

```
../fontforge/splinestroke.c: In function 'HandleJoin':
../fontforge/splinestroke.c:1878:6: warning: 'r_to' may be used uninitialized in this function [-Wmaybe-uninitialized]
 1878 |      IntersectLineCircle(p_fm, jpp->ut_fm, center_to, r_to, &i, &i2);
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../fontforge/splinestroke.c:1782:59: note: 'r_to' was declared here
 1782 |     bigreal t, K_fm, K_to, fsw_fm, fsw_to, fsw_avg, r_fm, r_to, i_dist;
      |                                                           ^~~~
../fontforge/splinestroke.c:2024:14: warning: 'r_fm' may be used uninitialized in this function [-Wmaybe-uninitialized]
 2024 |      ci_fm = ArcClip(center_fm, r_fm, neg_fm, p_fm, i, clip_ratio);
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../fontforge/splinestroke.c:1782:53: note: 'r_fm' was declared here
 1782 |     bigreal t, K_fm, K_to, fsw_fm, fsw_to, fsw_avg, r_fm, r_to, i_dist;
      |                                                     ^~~~
```

It might go away if using explicit booleans for K_fm==0 / K_to == 0?